src/nvim/api/buffer.c

```diff
@@ -291,7 +291,11 @@ void buffer_set_line_slice(Buffer buffer,
   // Only adjust marks if we managed to switch to a window that holds
   // the buffer, otherwise line numbers will be invalid.
   if (save_curbuf == NULL) {
-    mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
+    /// WCONV: extra is ssize_t, representing lines added or removed, so
+    /// WCONF: logically it should be < MAXLNUM, which is INT32_MAX
+    /// WCONF: Add an assertion to be sure
+    assert( -MAXLNUM <= extra && extra <= MAXLNUM);
```

Comment:
`ssize_t` is, by definition, not supposed to have any other negative value than `-1`. In practice, that's not how the type is implemented but if extra takes more negatives in a way that we need to test it like that, that's not the right type to use. I am no big shot but I think it's reasonable to change extra to `int` (or `linenr_T` if we want to keep it but I don't know the plan for that).